Src/BootCamp.Chapter/Program.cs

```diff
@@ -7,28 +7,48 @@ class Program
     {
         static void Main(string[] args)
         {
-            string userName = "", userSurname = "";
-            int userAge = 0;
-            float userWeight = 0f, userHeight = 0f;
+            CheckBMI("Tom", "Jefferson", 19, 50, 156.5f);
+            CheckBMI();
+        }
+
+        static void CheckBMI()
+        {
+            string userName;
+            CheckForUserInput("Please input your name.", out userName);
 
-            CheckForUserInput("Please input your name.", ref userName);
-            CheckForUserInput("Please input your surname.", ref userSurname);
+            string userSurname;
+            CheckForUserInput("Please input your surname.", out userSurname);
 
             Console.WriteLine("Hi " + userName + " " + userSurname + "!");
 
-            CheckForUserInput("Please input your age.", ref userAge);
+            int userAge;
+            CheckForUserInput("Please input your age.", out userAge);
 
-            CheckForUserInput("Please input your weight in kg.", ref userWeight);
-            CheckForUserInput("Please input your height in cm.", ref userHeight);
+            float userWeight;
+            CheckForUserInput("Please input your weight in kg.", out userWeight);
+            float userHeight;
+            CheckForUserInput("Please input your height in cm.", out userHeight);
+
+            CalculateBMI(userName, userSurname, userAge, userWeight, userHeight);
+        }
 
-            Console.WriteLine("Calculating BMI for " + userName + "...");
+        static void CheckBMI(in string userName, in string userSurname, in int userAge, in float userWeight, in float userHeight)
```

Comment:
Do not use 'in' keyword. By definition function parameters are inputs. Unless you intentionally wanted to pass everything by reference, which would be incorrect here.